src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASE_CAPACITY = 10;
+    private static final String INDEX_EXCEPTION_MESSAGE = "Provided index "
+            + "is < 0 or > list size. Please, provide valid index.";
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        this.objects = new Object[BASE_CAPACITY];
+    }
+
+    private void checkFreeSpace(int neededSpace) {
+        if (neededSpace > objects.length) {
+            int newLength = (int) (objects.length * 1.5);
+            if (newLength < neededSpace) {
+                newLength = neededSpace;
+            }
+            Object[] newArray = new Object[newLength];
+            System.arraycopy(objects, 0, newArray, 0, objects.length);
+            objects = newArray;
```

Comment:
The name says that this method should only check if there is a free space in the array, it should not change the array somehow. Please, refactor either the method name or the extract this logic somewhere else